### PR TITLE
[Core] Second pass at privatizing APIs.

### PR DESCRIPTION
**Pull request by clarkzinzow**
A second pass at privatizing APIs. This PR purely focuses on moving no-brainer modules to under the `_private` directory.

Each commit contains a single moved module (except for `ray_perf` and `ray_cluster_perf`, which were trivial and therefore grouped into a single commit), and I made sure to run the linter and rebuild + test using `test_basic.py` after each commit to catch obvious refactoring bugs.

## Related issue number

Towards #13995 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/node/node_head.py
Comment (rkooo567) [inline dashboard/modules/node/node_head.py:9]: Is this used?

**Review by wuisawesome — APPROVED** [review]

File: python/ray/autoscaler/_private/load_metrics.py
Comment (rkooo567) [inline python/ray/autoscaler/_private/load_metrics.py:13]: @wuisawesome Would this impact any backward compatibility of monitor?

File: dashboard/modules/node/node_head.py
Comment (clarkzinzow) [reply dashboard/modules/node/node_head.py:9]: Hmm not sure how this got here 🤔 I'll remove it

**Comment by rkooo567** [comment]
The goal is to move everything under ray/ to _private except worker.py -> api.py, am I correct?

File: dashboard/modules/node/node_head.py
Comment (clarkzinzow) [reply dashboard/modules/node/node_head.py:9]: Yep it's used below, [here](https://github.com/ray-project/ray/blob/e988fa9470cbd76db8858783eb871f899ebc870c/dashboard/modules/node/node_head.py#L249). Not sure how it wasn't failing before.

File: python/ray/autoscaler/_private/load_metrics.py
Comment (wuisawesome) [reply python/ray/autoscaler/_private/load_metrics.py:13]: Hmmm good catch. Let me investigate before we merge.

**Comment by clarkzinzow** [comment]
@rkooo567 That's the long-term goal, this PR is just moving some no-brainer modules. Getting to that long-term goal is going to be very incremental.

**Comment by rkooo567** [comment]
@clarkzinzow Yep! Agree with that we should split! Just would like to understand if we have the same view for the final state. If this LGTM if the monitor doesn't have any backward compatibility issue

File: python/ray/autoscaler/_private/load_metrics.py
Comment (clarkzinzow) [reply python/ray/autoscaler/_private/load_metrics.py:13]: I don't think that this would? Old autoscalers will use the old version of the Ray lib with `ray.gcs_utils` still being valid, and new autoscalers will use the new version of the Ray lib with `ray._private.gcs_utils` being valid. Nothing should change about the cluster <--> autoscaler interaction.

Lmk if I'm missing something there.

Comment (wuisawesome) [reply python/ray/autoscaler/_private/load_metrics.py:13]: ok, this should be fine.

**Review by rkooo567 — APPROVED** [review]

**Review by ijrsvt — APPROVED** [review]
LGTM, mostly just looked at my CODEOWNERS file. It overall looks good :)

**Comment by wuisawesome** [comment]
Looks like there are a few merge conflicts + legitimate test failures.

**Review by architkulkarni — APPROVED** [review]
Checked my codeowners file `test_snapshot.py`, looks good!

**Comment by clarkzinzow** [comment]
Currently blocked by https://github.com/ray-project/ray/pull/17912, I'll rebase this PR once that's merged and at that point the MacOS tests should pass.

**Comment by clarkzinzow** [comment]
@ericl Your commit broke lint, and I'm trying to keep the commits self-contained in case we decide to revert them, so I've been fixup-ing and rebasing.

**Comment by clarkzinzow** [comment]
Fixed

**Pull request merged**
